oauth2/handler.go

```diff
@@ -1021,6 +1021,32 @@ func (h *Handler) oauth2TokenExchange(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if accessRequest.GetGrantTypes().ExactOne(string(fosite.GrantTypeRefreshToken)) {
```

Comment:
Maybe a short comment explaining why we do this here?

Reply:
```suggestion
	// When refreshing tokens, we want to ensure to use the latest key-id available for signing the
	// potentially JWT-formatted tokens.
	if accessRequest.GetGrantTypes().ExactOne(string(fosite.GrantTypeRefreshToken)) {
```